### PR TITLE
tmuxPlugins: removed phony dependencies attribute

### PR DESCRIPTION
**Pull request by timstott**
###### Motivation for this change

The `dependencies` attribute in mkTmuxPlugin has deceived contributors
into thinking it makes packages available to the plugin at runtime.

To avoid further confusion the attribute is removed and plugins that
used it fixed.

Example PRs where `dependencies` has confused.
- https://github.com/NixOS/nixpkgs/pull/113726#discussion_r591146399
- https://github.com/NixOS/nixpkgs/pull/95275/files#r522817362

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by timstott** [comment]
@bhipple , @pjan, @ceedubs, @KyleOndy, @kidd :wave: I thought you might be interested by this change.

File: pkgs/misc/tmux-plugins/default.nix [hunk 1]
```diff
@@ -156,7 +152,15 @@ in rec {
       sha256 = "0gp37m3d0irrsih96qv2yalvr1wmf1n64589d4qzyzq16lzyjcr0";
       fetchSubmodules = true;
     };
-    dependencies = [ pkgs.gawk ];
+    nativeBuildInputs = [pkgs.makeWrapper];
```
Comment (SuperSandro2000) [inline pkgs/misc/tmux-plugins/default.nix:155]:
```suggestion
    nativeBuildInputs = [ pkgs.makeWrapper ];
```

File: pkgs/misc/tmux-plugins/default.nix [hunk 2]
```diff
@@ -156,7 +152,15 @@ in rec {
...
+    postInstall = ''
+      for f in config.sh tmux-fingers.sh setup-fingers-mode-bindings.sh; do
+      wrapProgram $target/scripts/$f \
+        --prefix PATH : ${with pkgs; lib.makeBinPath (
+          [gawk] ++ lib.optionals stdenv.isDarwin [reattach-to-user-namespace]
```
Comment (SuperSandro2000) [inline pkgs/misc/tmux-plugins/default.nix:160]:
```suggestion
          [ gawk ] ++ lib.optionals stdenv.isDarwin [ reattach-to-user-namespace ]
```

**Comment by bhipple** [comment]
+1 on the idea itself, always good to delete dead/confusing/useless code and slim down the implementation.

**Comment by timstott** [comment]
@SuperSandro2000 & @bhipple, I've added a deprecation error. Let me know what you think :raised_hands:

**Review by ethancedwards8 — APPROVED** [review]
Not sure why I got requested, but everything makes sense to me.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 118034` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
34 packages built:
</summary>
<ul>
<li>tmuxPlugins.battery</li>
<li>tmuxPlugins.continuum</li>
<li>tmuxPlugins.copycat</li>
<li>tmuxPlugins.cpu</li>
<li>tmuxPlugins.ctrlw</li>
<li>tmuxPlugins.dracula</li>
<li>tmuxPlugins.fingers</li>
<li>tmuxPlugins.fpp</li>
<li>tmuxPlugins.fzf-tmux-url</li>
<li>tmuxPlugins.gruvbox</li>
<li>tmuxPlugins.jump</li>
<li>tmuxPlugins.logging</li>
<li>tmuxPlugins.maildir-counter</li>
<li>tmuxPlugins.net-speed</li>
<li>tmuxPlugins.nord</li>
<li>tmuxPlugins.onedark-theme</li>
<li>tmuxPlugins.online-status</li>
<li>tmuxPlugins.open</li>
<li>tmuxPlugins.pain-control</li>
<li>tmuxPlugins.plumb</li>
<li>tmuxPlugins.power-theme</li>
<li>tmuxPlugins.prefix-highlight</li>
<li>tmuxPlugins.resurrect</li>
<li>tmuxPlugins.sensible</li>
<li>tmuxPlugins.sessionist</li>
<li>tmuxPlugins.sidebar</li>
<li>tmuxPlugins.sysstat</li>
<li>tmuxPlugins.tilish</li>
<li>tmuxPlugins.tmux-colors-solarized</li>
<li>tmuxPlugins.tmux-fzf</li>
<li>tmuxPlugins.urlview</li>
<li>tmuxPlugins.vim-tmux-focus-events</li>
<li>tmuxPlugins.vim-tmux-navigator</li>
<li>tmuxPlugins.yank</li>
</ul>
</details>

**Pull request merged**